vm-virtio/src/vhost_user/fs.rs

```diff
@@ -309,6 +319,16 @@ impl VirtioDevice for Fs {
             };
         self.kill_evt = Some(self_kill_evt);
 
+        let (self_pause_evt, pause_evt) =
+            match EventFd::new(EFD_NONBLOCK).and_then(|e| Ok((e.try_clone()?, e))) {
```

Comment:
I see this in every device. Wouldn't a `map_err()` be more concise?

Reply:
Indeed. There's a *lot* of duplicated code in the activation routines for all our virtio devices. We're creating a `kill_evt` `EventFd` pair for each and every one of them, and I made both EventFd pair creations (pause and kill) use a `map_err`.